sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -532,6 +534,102 @@ public String getId() {
         return withContext(context -> readItemInternal(itemId, requestOptions, itemType, context));
     }
 
+    /**
+     * Reads many documents.
+     *
+     * @param <T> the type parameter
+     * @param itemKeyList document id and partition key pair that needs to be read
+     * @param classType   class type
+     * @return a Mono with feed response of cosmos items
+     */
+    @Beta(Beta.SinceVersion.V4_4_0)
+    public <T> Mono<FeedResponse<T>> readMany(
+        List<Pair<String, PartitionKey>> itemKeyList,
+        Class<T> classType) {
+
+        return this.readMany(itemKeyList, null, classType);
+    }
+
+    /**
+     * Reads many documents.
+     *
+     * @param <T> the type parameter
+     * @param itemKeyList document id and partition key pair that needs to be read
+     * @param sessionToken the optional Session token - null if the read can be made without specific session token
+     * @param classType   class type
+     * @return a Mono with feed response of cosmos items
+     */
+    @Beta(Beta.SinceVersion.V4_4_0)
+    public <T> Mono<FeedResponse<T>> readMany(
+        List<Pair<String, PartitionKey>> itemKeyList,
+        String sessionToken,
```

Comment:
Is Itemizing (or promoting parts of) request option, to limit to applicable ones?

Reply:
Yes - the only relevant one is SessionToken compared to CosmosQueryRequestOptions